tests/phpunit/tests/test-razorpay-route.php

```diff
@@ -565,4 +565,117 @@ function () {
 
         $this->assertSame(date("d M Y h:i A", strtotime('+5 hour +30 minutes', 1677542400)), $response[0]['created_at']);
     }
+
+    public function testrzpRoutePayments()
+    {
+        $this->instance->shouldReceive('routeHeader');
+
+        $this->instance->shouldReceive('preparePaymentItems');
+
+        $this->instance->shouldReceive('search_box')->with('search', 'search_id');
+
+        $this->instance->shouldReceive('display');
+
+        ob_start();
+        $this->instance->rzpRoutePayments();
+        $result = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertStringContainsString('<div class="wrap route-container"><form method="get">',$result);
+
+        $this->assertStringContainsString('<input type="hidden" name="page" value="razorpayRoutePayments">',$result);
+
+        $this->assertStringContainsString('<p class="pay_search_label">Search here for payments of linked account</p>',$result);
+    }
+
+    public function testpreparePaymentItems()
+    {
+        $this->instance->shouldReceive('get_pagenum')->andReturn(12);
+
+        $_REQUEST = array('s' => 'ABC123');
+
+        $paymentItems = array(
+            'payment_id' => 'pay_LEkixKpTE1Mvrk',
+            'order_id' => 11,
+            'amount' => '2500',
+            'email' => 'abc.xyz@razorpay.com',
+            'contact' => '0987654321',
+            'created_at' => '16/02/2023',
+            'status' => 'Pending');
+
+        $this->instance->shouldReceive('getPaymentItems')->with(10, 'ABC123')->andReturn($paymentItems);
+
+        $this->instance->shouldReceive('set_pagination_args');
+
+        $this->instance->preparePaymentItems();
+
+        $this->assertTrue(true);
+    }
+
+    public function testpreparePaymentItemswithouroffset()
+    {
+        $this->instance->shouldReceive('get_pagenum')->andReturn(0);
+
+        $_REQUEST = array('s' => 'ABC123');
+
+        $paymentItems = array(
+            'payment_id' => 'pay_LEkixKpTE1Mvrk',
```

Comment:
This testcase is getting failed due to undefined array key 0
Add array key 0 to $paymentItems array